toolchain/check/modifiers.cpp

```diff
@@ -88,7 +88,12 @@ auto CheckAccessModifiersOnDecl(Context& context, Lex::TokenKind decl_kind)
 
 auto RequireDefaultFinalOnlyInInterfaces(Context& context,
                                          Lex::TokenKind decl_kind) -> void {
-  // TODO: Skip this if *context.current_scope_kind() == SemIR::InterfaceDecl
+  if (auto kind = context.current_scope_kind()) {
+    if (*kind == SemIR::InterfaceDecl::Kind) {
+      // Both `default` and `final` allowed in an interface definition.
+      return;
+    }
+  }
```

Comment:
Maybe you could simplify this to:
```suggestion
  if (context.GetCurrentScopeAs<SemIR::InterfaceDecl>()) {
    // Both `default` and `final` allowed in an interface definition.
    return;
  }
```
(I think this ends up doing slightly more work, but I'd expect it to optimize to the same thing.)

Or if not that, then maybe we could add a `CurrentScopeIs<Kind>`? It looks like the other caller of `current_scope_kind` would also benefit from this change.

Reply:
Done.